llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -267,6 +262,13 @@ static void collectKernelModuleMap(
         ResKernelModuleMap["<GLOBAL>"].push_back(&F);
         break;
       }
+    } else if (EntryScope == Scope_PerModule &&
+               F.getCallingConv() == CallingConv::SPIR_KERNEL) {
```

Comment:
```suggestion
    } else if (EntryScope == Scope_PerModule) {
```

Reply:
No, I actually cannot do this, since `spir_func` are allowed not to have `sycl-module-id` info. Those are the functions we need to drop.

Reply:
I see. This logic is important for understanding - please add a comment then.

Reply:
Done